src/esperanza/validator.h

```diff
@@ -21,8 +21,17 @@ class Validator {
 
   uint160 m_validator_address;
   uint64_t m_deposit;
+
+  // At which dynasty finalizer can start voting.
   uint32_t m_start_dynasty;
+
+  // At which dynasty finalizer stops voting. However, finalizer will actually
+  // stop at m_end_dynasty + 1 because it allowed to vote when it is active
```

Comment:
"because it allowed to vote" or "because it is allowed to vote"?

Besides that, I agree with @frolosofsky on the fact that this text could be somewhat misleading.

Reply:
I removed this description and wrote one sentence only. Please, re-open the conversion if further changes needed.